dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java

```diff
@@ -23,95 +18,54 @@ public class CacheConfigManager {
 
     private static final CacheConfigManager CACHE_CONFIG_MANAGER = new CacheConfigManager();
 
-    private static LoadingCache<String, String> installationAccessTokenCache;
+    private static LoadingCache<Long, GitHub> githubClientAPICache;
 
-    private static volatile String jsonWebToken;
+    private static String appId;
```

Comment:
I think this and the next line, and probably line 21 as well, should technically be volatile, which may also answer your question about why `jsonWebToken` was declared volatile -- if the Dropwizard startup thread writes to this value, there's no guarantee that another thread servicing a request will read the most recent value unless you have `volatile`.

In practice, the other thread almost always does see it.  I remember trying to reproduce the theoretical hang described [here](https://www.baeldung.com/java-volatile), and was never able to locally, but it's supposed to happen sometimes.

Reply:
Hmmm, this isn't exactly the same as before though. 
This is now an `appId` and this should only be written once ever since `initCache` is only called once from the start-up thread which I assume is only done in one thread. 

On the other hand, the jsonWebToken is now inside `JWTTokenProvider` which is nested inside the `GitHub` instances in the cache, but `GitHub` claims to be thread-safe.

Reply:
Hmm, looking at it more closely, it does look like `appId` and `privateKeyFile` are written to/read in the same thread, so my concurrency concern was off-base.

Given that, do you even need to save them as fields? It looks like you could just pass them down from `initCache` as parameters to `getGitHubClientFromInstallationId`?